Data/Hashable/Class.hs

```diff
@@ -591,7 +601,7 @@ instance Hashable1 [] where
         step (SP s l) x   = SP (h s x) (l + 1)
 
 instance Hashable B.ByteString where
-    hashWithSalt salt bs = B.inlinePerformIO $
+    hashWithSalt salt bs = unsafeDupablePerformIO $
```

Comment:
Is this really performance equivalent? Could you run the benchmarks with and without this patch?